lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -157,13 +151,7 @@ defmodule KinesisClient.Stream.Shard.Producer do
   def handle_info({:ack, _ref, successful_msgs, failed_msgs}, state) do
     %{metadata: %{"SequenceNumber" => checkpoint}} = successful_msgs |> Enum.reverse() |> hd()
 
-    :ok =
-      AppState.update_checkpoint(
-        state.shard_id,
-        state.lease_owner,
-        checkpoint,
-        state.app_state_opts
-      )
+    :ok = update_checkpoint(state, checkpoint)
```

Comment:
This change actually fixes a separate bug in the update checkpoint flow. The previous call to `update_checkpoint/5` here was actually invalid (it's missing the `state.app_name` as the first arg)!

This was missed previously b/c none of the unit tests covered this function head. I've added a unit test for this case!